Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift

```diff
@@ -101,12 +134,33 @@ open class ChatChannelAvatarView: _View, ThemeProvider, SwiftUIRepresentable {
         
         // We show a combination of at max 4 images combined
         urls = Array(urls.prefix(maxNumberOfImagesInCombinedAvatar))
-        
+        let customKeyForCache = urls.compactMap({ $0?.lastPathComponent}).joined()
+        // Checked if avatar already cached
+        if let imageContainer = ImagePipeline.shared.cache.cachedImage(for: customKeyForCache), let cachedUrlCount = imageContainer.userInfo["count"] as? Int {
+            if urls.count == cachedUrlCount{
+                loadIntoAvatarImageView(from: nil, placeholder: imageContainer.image)
```

Comment:
Check for character count linting here. 
Also update `if urls.count == cachedUrlCount` with the above if let only.